src/parser.rs

```diff
@@ -32,21 +33,18 @@ pub fn kill_node_attribute(node: &NixNode) -> Result<NixNode, String> {
                 .green()
                 .children()
                 .enumerate()
-                // TODO better error handling
                 .filter_map(|(idx, val)| {
+                    // the '.to_owned()' is required to turn GreenNodeData into GreenNode
+                    // because GreenNodeData doesn't implement PartialEq
                     val.into_node().and_then(|inner_node| {
-                        // the '.to_owned()' is required to turn GreenNodeData into GreenNode
-                        // because GreenNodeData doesn't implement PartialEq
-                        if *inner_node == node.green().to_owned() {
-                            Some(idx)
-                        } else {
-                            None
-                        }
+                        (*inner_node == node.green().to_owned()).then(|| idx)
                     })
                 })
-                // TODO is this really what we want
                 .last()
-                .unwrap_or(0);
+                .ok_or(Err::<NixNode, String>(
+                    "node not in parent tree".to_string(),
+                ))
+                .unwrap();
```

Comment:
why? also, why do you wrap that into `Err::<NixNode, String>` and not just `                .ok_or("node not in parent tree")` or `.expect("node not in parent tree")`?

Reply:
`expect` is definitely cleaner. Changing to that. It might even be worthwhile switching collecting to a vec and asserting the length is 1.

The child node (`node`) should be in the list of parent node's children exactly once. If the child doesn't exist in the parent node's children set, the function should fail and return an error since the AST is in an inconsistent state.

Reply:
I switched to an assert that the vec length is one instead.